lib/jnpr/junos/utils/sw.py

```diff
@@ -960,26 +960,40 @@ def reboot(self, in_min=0, at=None, all_re=True):
         :param bool all_re: In case of dual re or VC setup, function by default
             will reboot all. If all is False will only reboot connected device
 
+        :param str on_node: In case of linux based device, function will by default
+            reboot the whole device. If any specific node is mentioned,
+            reboot will be performed on mentioned node
+
         :returns:
             * reboot message (string) if command successful
 
         :raises RpcError: when command is not successful.
 
         .. todo:: need to better handle the exception event.
         """
+        if self._dev.facts['_is_linux']:
+            if on_node is None:
+                cmd = E('request-shutdown-reboot')
+            else:
+                cmd = E('request-node-reboot')
+                cmd.append(E('node', on_node))
+        else:
+            cmd = E('request-reboot')
+            if all_re is True:
+                if self._multi_RE is True and self._multi_VC is False:
+                    cmd.append(E('both-routing-engines'))
+                elif self._mixed_VC is True:
+                    cmd.append(E('all-members'))
         if in_min >= 0 and at is None:
-            cmd = E('request-reboot', E('in', str(in_min)))
+            cmd.append(E('in', str(in_min)))
         else:
-            cmd = E('request-reboot', E('at', str(at)))
-
-        if all_re is True:
-            if self._multi_RE is True and self._multi_VC is False:
-                cmd.append(E('both-routing-engines'))
-            elif self._mixed_VC is True:
-                cmd.append(E('all-members'))
+            cmd.append(E('at', str(at)))
         try:
             rsp = self.rpc(cmd)
-            got = rsp.getparent().findtext('.//request-reboot-status').strip()
+            if self._dev.facts['_is_linux']:
+                got = rsp.text
+            else:
+                got = rsp.getparent().findtext('.//request-reboot-status').strip()
```

Comment:
@rsmekala Have you tested that for this new case too rpc reply will have request-reboot-status tag?